File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.cpp
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
vector<vector<int>>ans;
if(root==NULL) return ans;
queue<TreeNode*>q;
q.push(root);
while(q.size())
{
    int n=q.size();
    vector<int>temp;
    for(int i=0 ; i<n ; i++)
    {
        TreeNode* current=q.front();
        q.pop();
        if(current->left) q.push(current->left);
        if(current->right) q.push(current->right);
        temp.push_back(current->val);
    }
    ans.push_back(temp);
}
return ans;
    }
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>>ans=new LinkedList<List<Integer>>();
        Queue<TreeNode>q=new LinkedList<TreeNode>();
        
        if(root==null) return ans;
        q.offer(root);
        while(q.size()!=0)
        {
            int n=q.size();
            List<Integer>temp=new ArrayList<>();
            for(int i=0 ; i<n ; i++)
            {
                TreeNode curr=q.peek();
                q.poll();
                if(curr.left!=null) q.offer(curr.left);
                if(curr.right!=null) q.offer(curr.right);
                temp.add(curr.val);
            }
            ans.add(temp);
        }
        return ans;
    }
}